projects/igniteui-angular/src/lib/action-strip/grid-actions/grid-editing-actions.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, HostBinding } from '@angular/core';
+import { IgxGridActionsBaseDirective } from './grid-actions-base.directive';
+
+@Component({
+    selector: 'igx-grid-editing-actions',
+    templateUrl: 'grid-editing-actions.component.html',
+    providers: [{ provide: IgxGridActionsBaseDirective, useExisting: IgxGridEditingActionsComponent }]
+})
+
+export class IgxGridEditingActionsComponent extends IgxGridActionsBaseDirective {
+    /**
+     * Host `class.igx-action-strip` binding.
+     * @hidden
+     * @internal
+     */
+    @HostBinding('class.igx-action-strip__editing-actions')
+    public cssClass = 'igx-action-strip__editing-actions';
+
+    /**
+     * Enter row or cell edit mode depending the grid rowEditable option
+     * @example
+     * ```typescript
+     * this.gridEditingActions.startEdit();
+     * ```
+     */
+    public startEdit(): void {
+        const context = this.strip.context;
```

Comment:
Do type the context here - this will ensure all the calls below use traceable API :)
Can even use a check since it's public API:
```
if (!(this.strip.context instanceof IgxRowDirective)) {
            return;
        }
```
which will actually type the code afterwards. You can make your base class check a custom type guard too - see https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
This applies for all action components